### DISCUSSION TITLE
weblog reason reset code

### DISCUSSION THREAD

**Discussion by danilkorotkov** [HomeSpan/HomeSpan#530] [Ideas]
it is better to use the names from the Espressif code
```
case 1  : Serial.println ("Vbat power on reset");break;
    case 3  : Serial.println ("Software reset digital core");break;
    case 4  : Serial.println ("Legacy watch dog reset digital core");break;
    case 5  : Serial.println ("Deep Sleep reset digital core");break;
    case 6  : Serial.println ("Reset by SLC module, reset digital core");break;
    case 7  : Serial.println ("Timer Group0 Watch dog reset digital core");break;
    case 8  : Serial.println ("Timer Group1 Watch dog reset digital core");break;
    case 9  : Serial.println ("RTC Watch dog Reset digital core");break;
    case 10 : Serial.println ("Instrusion tested to reset CPU");break;
    case 11 : Serial.println ("Time Group reset CPU");break;
    case 12 : Serial.println ("Software reset CPU");break;
    case 13 : Serial.println ("RTC Watch dog Reset CPU");break;
    case 14 : Serial.println ("for APP CPU, reseted by PRO CPU");break;
    case 15 : Serial.println ("Reset when the vdd voltage is not stable");break;
    case 16 : Serial.println ("RTC Watch dog reset digital core and rtc module");break;
    default : Serial.println ("NO_MEAN");
```

**Comment by HomeSpan** [comment]
When I originally added the code I noticed that Espressif has changed the meanings over time.  Did not want to hard code them.

**Reply by danilkorotkov** [reply]
does it depend on the IDF version?

**Comment by HomeSpan** [comment]
It did vary depending on the IDF, but since version 4 it seems to have become stable.   Note I am using [esp_reset_reason()](https://docs.espressif.com/projects/esp-idf/en/v5.0.1/esp32/api-reference/system/misc_system_api.html#_CPPv416esp_reset_reasonv), not `rtc_get_reset_reason()` (which you are referencing above and contains a different set of enumerations).  The `esp_reset_reason()` function is not CPU specific, and most importantly,  contains a response  code for resetting due to a Panic, which is often helpful to know.
